dashboard/src/utils/date.ts

```diff
@@ -8,3 +8,41 @@ export const dateObjectToTimestampInSeconds = (date: Date): number => {
 export const daysToSeconds = (days: number): number => {
   return days * SECONDS_IN_ONE_DAY;
 };
+
+const isValidDate = (date: Date): boolean => {
+  return !isNaN(date.getTime());
+};
+
+export const getFormattedDate = (
+  timestampInSeconds: number,
+): string | undefined => {
+  const date = new Date(timestampInSeconds * MILLISECONDS_IN_ONE_SECOND);
+  if (!isValidDate(date)) {
+    return undefined;
+  }
+
+  const dateResult = date.toLocaleDateString('en-US', {
+    year: 'numeric',
+    month: 'long',
+    day: 'numeric',
+  });
+
+  return dateResult;
+};
+
+export const getFormattedTime = (
+  timestampInSeconds: number,
+): string | undefined => {
+  const date = new Date(timestampInSeconds * MILLISECONDS_IN_ONE_SECOND);
+  if (!isValidDate(date)) {
+    return undefined;
+  }
+
+  const timeResult = date.toLocaleTimeString('en-US', {
+    hour: '2-digit',
+    minute: '2-digit',
+    second: '2-digit',
+  });
+
+  return timeResult;
+};
```

Comment:
we can unify this in a function.
```suggestion
export const getFormattedDatetime = (
  timestampInSeconds: number,
): string | undefined => {
  const date = new Date(timestampInSeconds * MILLISECONDS_IN_ONE_SECOND);
  if (!isValidDate(date)) {
    return undefined;
  }

  const dateResult = date.toLocaleDateString('en-US', {
    year: 'numeric',
    month: 'long',
    day: 'numeric',
    hour: '2-digit',
    minute: '2-digit',
    second: '2-digit',
  });

  return dateResult;
}
};
```

Reply:
I will leave separated because it is more flexible